File: src/java/NinjaSurvivalRate/src/test/java/ninjasurvivalrate/KillerSteps.java
package ninjasurvivalrate;

import cuke4duke.annotation.I18n.EN.*;
import cuke4duke.annotation.*;
import java.util.Map;
import java.util.List;
import java.util.Collection;
import static junit.framework.Assert.assertEquals;

public class KillerSteps {

 //START:steps 
 private static int killed=0;
 private static int remaining=3;

 @Before 
 public void killNinjaBeforeEachScenario(){
   killed++;
 }
 @After
 public void expectLessNinjasAfterScenario(){
   remaining--;
 }
 @When ("^this scenario is executed$")
 public void thisScenarioIsExecuted() {
  // do absolutely nothing
 }
 @Then ("^Chuck Norris should expect ([0-9]+)* ninjas$")
 public void chuckNorrisShouldExpectNinjas(int expected) {
   assertEquals(expected, remaining);
 }
 @Then ("^Chuck Norris should kill one ninja$")
 public void chuckNorrisShouldKillOneNinja() {
   heShouldKillNinjas(1);
 }
 @Then ("^he should kill ([0-9]+)* ninjas$")
 public void heShouldKillNinjas(int expected) {
   assertEquals(expected, killed);
 }
 //END:steps
}

File: src/dotnet/NinjaSurvivalRate/src/KillerSteps.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Cuke4Nuke.Framework;
using NUnit.Framework;

namespace NinjaSurvivalRate
{
    public class KillerSteps
    {
        //START:steps 
        private static int killed = 0;
        private static int remaining = 3;

        [Before]
        public void KillNinjaBeforeEachScenario()
        {
            killed++;
        }

        [After]
        public void ExpectLessNinjasAfterScenario()
        {
            remaining--;
        }

        [When(@"^this scenario is executed$")]
        public void ThisScenarioIsExecuted()
        {
            // do absolutely nothing
        }

        [Then("^Chuck Norris should expect ([0-9]+)* ninjas$")]
        public void ChuckNorrisShouldExpectNinjas(int expected)
        {
            Assert.AreEqual(expected, remaining);
        }

        [Then("^Chuck Norris should kill one ninja$")]
        public void ChuckNorrisShouldKillOneNinja()
        {
            HeShouldKillNinjas(1);
        }

        [Then("^he should kill ([0-9]+)* ninjas$")]
        public void HeShouldKillNinjas(int expected)
        {
            Assert.AreEqual(expected, killed);
        }

        //END:steps
    }
}